chore(CI): add a rebase gha for the mutations branch

josecorella opened the pull request:
### Issue #, if available:

### Description of changes:

### Squash/merge commit message, if applicable:

```
<type>(dafny/java/python/dotnet/go/rust): <description>
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

josecorella committed:
chore(CI): add a rebase gha for the mutations branch

josecorella committed:
format

imabhichow commented on .github/workflows/rebase-mutations.yml:
```diff
@@ -0,0 +1,65 @@
+        if: steps.rebase.outputs.rebase_status == 'success'
+        run: git push --force-with-lease origin mutations
+
+      - name: Create issue on rebase failure
+        if: steps.rebase.outputs.rebase_status == 'failed'
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            github.rest.issues.create({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              title: 'Rebase failed for mutations branch',
+              body: `The automatic rebase of the \`mutations\` branch failed due to conflicts.\n\n` +
+                    `This happened after a push to the default branch by ${context.actor}.\n\n` +
+                    `Please resolve the conflicts manually.`
+            })
```
Out of curiosity, do we create a new GHI for each commit that has conflicts or will it be added to an already cut GHI?

josecorella replied:
create a new one for each commit - highly overkill and will create lots of noise. We may opt for a different alert mechanism.

josecorella committed:
add feedback

josecorella committed:
Merge branch 'main' into jocorell/rebase-gha

texastony commented:
Question: Would we need to collapse from 3 PRs/branches to just one?

Commit:
ci: Prevent script injection in GitHub Actions workflows

Move user-influenced github.* context values out of inline run-block
interpolation and into env: variables, matching the remediation already
present on main. Resolves ACAT script-injection findings.

sim: https://t.corp.amazon.com/V2307997274
